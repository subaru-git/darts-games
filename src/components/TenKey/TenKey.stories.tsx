import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import TenKey from './TenKey';

export default {
  title: 'Components/TenKey',
  component: TenKey,
} as ComponentMeta<typeof TenKey>;

const Template: ComponentStory<typeof TenKey> = (args) => <TenKey {...args} />;

export const Default = Template.bind({});
Default.args = {};
