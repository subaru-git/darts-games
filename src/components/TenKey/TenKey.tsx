import React, { FC, useState } from 'react';
import { AiOutlineEnter } from 'react-icons/ai';
import { RiDeleteBack2Line } from 'react-icons/ri';
import Button from '@/atoms/Button';

type TenKeyProps = {
  onCount: (value: string) => void;
};

const TenKey: FC<TenKeyProps> = ({ onCount }) => {
  const [value, setValue] = useState('');
  return (
    <>
      <div className='grid grid-cols-3 grid-rows-5 gap-1'>
        <input
          type='text'
          className='col-span-3 rounded-md border p-2.5 text-right text-sm tracking-widest text-gray-900'
          value={value}
        />
        <Button onClick={() => setValue(value + '1')} aria-label='one'>
          1
        </Button>
        <Button onClick={() => setValue(value + '2')} aria-label='two'>
          2
        </Button>
        <Button onClick={() => setValue(value + '3')} aria-label='three'>
          3
        </Button>
        <Button onClick={() => setValue(value + '4')} aria-label='four'>
          4
        </Button>
        <Button onClick={() => setValue(value + '5')} aria-label='five'>
          5
        </Button>
        <Button onClick={() => setValue(value + '6')} aria-label='six'>
          6
        </Button>
        <Button onClick={() => setValue(value + '7')} aria-label='seven'>
          7
        </Button>
        <Button onClick={() => setValue(value + '8')} aria-label='eight'>
          8
        </Button>
        <Button onClick={() => setValue(value + '9')} aria-label='nine'>
          9
        </Button>
        <Button onClick={() => setValue(value.slice(0, -1))} aria-label='delete'>
          <RiDeleteBack2Line className='m-auto' />
        </Button>
        <Button onClick={() => setValue(value + '0')} aria-label='zero'>
          0
        </Button>
        <Button
          onClick={() => {
            onCount(value);
            setValue('');
          }}
          aria-label='enter'
        >
          <AiOutlineEnter className='m-auto' />
        </Button>
      </div>
    </>
  );
};

export default TenKey;
