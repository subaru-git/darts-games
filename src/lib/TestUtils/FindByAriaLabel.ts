export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
