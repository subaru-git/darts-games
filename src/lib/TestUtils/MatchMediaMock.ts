import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
