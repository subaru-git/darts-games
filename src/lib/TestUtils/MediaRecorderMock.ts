export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
