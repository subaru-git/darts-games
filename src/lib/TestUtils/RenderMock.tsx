import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
