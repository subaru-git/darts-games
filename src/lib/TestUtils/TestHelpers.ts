import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}