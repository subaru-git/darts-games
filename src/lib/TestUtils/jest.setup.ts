import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
