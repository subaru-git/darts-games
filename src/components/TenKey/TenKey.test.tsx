import TenKey from './TenKey';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const onCountMock = jest.fn();
  const { container } = render(<TenKey onCount={onCountMock} />);
  expect(container).toMatchSnapshot();
});
