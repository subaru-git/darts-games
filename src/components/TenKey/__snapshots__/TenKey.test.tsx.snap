// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="grid grid-cols-3 grid-rows-5 gap-1"
  >
    <input
      class="col-span-3 rounded-md border p-2.5 text-right text-sm tracking-widest text-gray-900"
      type="text"
      value=""
    />
    <button
      aria-label="one"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      1
    </button>
    <button
      aria-label="two"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      2
    </button>
    <button
      aria-label="three"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      3
    </button>
    <button
      aria-label="four"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      4
    </button>
    <button
      aria-label="five"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      5
    </button>
    <button
      aria-label="six"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      6
    </button>
    <button
      aria-label="seven"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      7
    </button>
    <button
      aria-label="eight"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      8
    </button>
    <button
      aria-label="nine"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      9
    </button>
    <button
      aria-label="delete"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      <svg
        class="m-auto"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M6.535 3H21a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1H6.535a1 1 0 0 1-.832-.445l-5.333-8a1 1 0 0 1 0-1.11l5.333-8A1 1 0 0 1 6.535 3zm.535 2l-4.666 7 4.666 7H20V5H7.07zM13 10.586l2.828-2.829 1.415 1.415L14.414 12l2.829 2.828-1.415 1.415L13 13.414l-2.828 2.829-1.415-1.415L11.586 12 8.757 9.172l1.415-1.415L13 10.586z"
          />
        </g>
      </svg>
    </button>
    <button
      aria-label="zero"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      0
    </button>
    <button
      aria-label="enter"
      class="text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50 rounded-lg border py-2 px-3 text-base font-bold subpixel-antialiased md:text-base"
    >
      <svg
        class="m-auto"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M864 170h-60c-4.4 0-8 3.6-8 8v518H310v-73c0-6.7-7.8-10.5-13-6.3l-141.9 112a8 8 0 0 0 0 12.6l141.9 112c5.3 4.2 13 .4 13-6.3v-75h498c35.3 0 64-28.7 64-64V178c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
